frontend/apps/reader/modules/readerpanning.lua

```diff
@@ -21,7 +21,7 @@ end
 function ReaderPanning:onGesture() end
 
 function ReaderPanning:registerKeyEvents()
-    if Device:hasKeyboard() then
+    if Device:hasKeyboard() and Device:hasDPad() and not Device:useDPadAsActionKeys() then
```

Comment:
Is the `Device:hasKeyboard()` still needed?

Reply:
Well i am not sure really. I suppose as there are devices with dpads but not enough keys (hasFewKeys)